rs/machine/handlers/handler.py

```diff
@@ -9,5 +9,5 @@ class Handler:
     def can_handle(self, state: GameState) -> bool:
         return False
 
-    def handle(self, state: GameState) -> HandlerAction:
+    def handle(self, state: GameState, slay_heart: bool) -> HandlerAction:
```

Comment:
So, this works for now, but I think we may want to adjust the approach here.

1. We're only telling it when it handles whether it should consider slaying the heart, but not on can_handle. So that means that we couldn't ever have a handler distinguish between whether or not it should be considered whether it's going for the heart or not. Ultimately, this means less reusable handlers.

2. It will be kind of ugly having this exist in both methods can_handle and handle of the handlers. Perhaps injection via constructor when we're making the handlers would be nicer. (Something to consider, not saying we _have_ to make that change).

Reply:
Agreed on not making further changes for now - there are bigger changes we'd want to make here.
Might make a suggestion for people to use latest release to avoid too many issues with us changing stuff all the time.